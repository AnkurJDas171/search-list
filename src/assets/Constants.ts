export const ALERT_MESSAGE = "This user name does not exist! Please specify an existing user name!";
export const ALERT_VISIBLE_TIME = 2000;

export const LIST_TITLES = [
    'Name',
    'Rank',
    'No. of bananas',
    'isSearchedUser?'
];

export const LIST_LENGTH = 10;
