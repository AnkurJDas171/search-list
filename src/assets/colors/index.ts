const colors = {
    // UI components
    BUTTON: 'rgba(111, 202, 186, 1)',
    TEXT_INPUT: 'rgba(243, 243, 243, 1)',
    LIST_TITLE: 'rgba(111, 202, 186, 1)',
    LIST_ODD_ITEM: 'rgba(204, 255, 229, 1)',
    LIST_EVEN_ITEM: 'rgba(236, 252, 245, 1)',
    SEARCH_ITEM: 'rgba(182, 189, 186, 1)',

    // color names
    BLACK: 'rgba(0, 0, 0, 1)',
    BLACK_SEMI_TRANSPARENT: 'rgba(0, 0, 0, 0.2)',
    WHITE: 'rgba(255, 255, 255, 1)',

    // type
    TRANSPARENT: 'rgba(255, 255, 255, 0)'
}

export default colors;
