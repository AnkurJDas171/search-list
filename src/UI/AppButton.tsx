import React from "react";
import { StyleSheet } from "react-native";
import { Button } from '@rneui/themed';

import { AppButtonProps } from "./types";
import { testID } from "../assets/Constants";
import { fontSize } from "../utils";

const AppButton = ({
    color,
    containerStyle,
    handlePress,
    isDisable,
    title,
    titleColor,
}: AppButtonProps): JSX.Element => {
    return (
        <Button
            title={title}
            buttonStyle={[styles.button, { backgroundColor: color }]}
            titleStyle={[styles.title, { color: titleColor }]}
            containerStyle={[styles.container, containerStyle]}
            onPress={handlePress}
            disabled={isDisable}
            testID={testID.APP_BUTTON}
        />
    )
}

const styles = StyleSheet.create({
    button: {
        borderRadius: 5
    },
    container: {
        paddingVertical: 5,
        width: '30%'
    },
    title: {
        fontWeight: 'bold',
        fontSize: fontSize(11)
    }
})

export default AppButton;
