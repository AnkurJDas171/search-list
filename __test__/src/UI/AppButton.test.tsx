import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react-native";

import AppButton from "../../../src/UI/AppButton"
import { testID } from "../../../src/assets/Constants";

describe('App button', () => {
    it('should render button', async () => {
        const screen = render(
            <AppButton
                color={"#000"}
                handlePress={() => { }}
                isDisable={false}
                title={"search"}
                titleColor={"#fff"}
            />
        );

        expect(screen.toJSON()).toMatchSnapshot();

        const button = await screen.findByTestId(testID.APP_BUTTON);
        expect(button).not.toBeNull();
    });

    it('button press should work', async () => {
        const handlePress = () => {
            console.log("app button pressed");
        }
        jest.spyOn(console, 'log');

        const screen = render(
            <AppButton
                color={"#000"}
                handlePress={handlePress}
                isDisable={false}
                title={"search"}
                titleColor={"#fff"}
            />
        );

        const button = await screen.findByTestId(testID.APP_BUTTON);
        fireEvent.press(button);

        await expect(console.log).toHaveBeenCalledWith("app button pressed");
    })
})