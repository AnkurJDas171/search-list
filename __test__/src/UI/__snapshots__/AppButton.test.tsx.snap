// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App button should render button 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      {
        "borderRadius": 2,
      },
      [
        {
          "paddingVertical": 5,
          "width": "30%",
        },
        undefined,
      ],
      false,
    ]
  }
  testID="RNE_BUTTON_WRAPPER"
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": false,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="app-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#000",
          "borderColor": "#2089dc",
          "borderRadius": 5,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "bold",
            "paddingVertical": 1,
            "textAlign": "center",
          }
        }
      >
        search
      </Text>
    </View>
  </View>
</View>
`;
